source/state_representation/test/tests/space/cartesian/test_cartesian_state.cpp

```diff
@@ -0,0 +1,128 @@
+#include <gtest/gtest.h>
+
+#include "state_representation/space/cartesian/CartesianState.hpp"
+#include "state_representation/exceptions/NotImplementedException.hpp"
+
+using namespace state_representation;
+
+TEST(CartesianStateTest, IdentityInitialization) {
+  CartesianState identity = CartesianState::Identity("test");
+  // the joint state should not be considered empty (as it is properly initialized)
+  EXPECT_FALSE(identity.is_empty());
+  // all data should be zero except for orientation that should be identity
+  EXPECT_EQ(identity.get_position().norm(), 0);
+  EXPECT_EQ(identity.get_orientation().norm(), 1);
+  EXPECT_EQ(identity.get_orientation().w(), 1);
+  EXPECT_EQ(identity.get_twist().norm(), 0);
+  EXPECT_EQ(identity.get_accelerations().norm(), 0);
+  EXPECT_EQ(identity.get_wrench().norm(), 0);
+}
+
+TEST(CartesianStateTest, RandomStateInitialization) {
+  CartesianState random = CartesianState::Random("test");
+  // all data should be random (non 0)
+  EXPECT_GT(random.get_position().norm(), 0);
```

Comment:
I would prefer `EXPECT_NE` in place of `EXPECT_GT` in all these cases. It is more accurate to what you actually expect (per the comment above), and will save you calling `abs` everywhere